src/Geometry/MeshReader.h

```diff
@@ -278,6 +289,89 @@ class MeshReader
 		}
 	}
 
+	void exchangeVerticesWithMPINeighbors() {
```

Comment:
It can happen that +/- fault elements belong to different partitions. Nevertheless we need to evaluate basis functions for + and - side. To achieve this, we need to have tetrahedrons from both sides. Instead of exchanging +/- tetrahedrons only along fault surfaces, the Fortran code exchanges all halo tetrahedrons between neighbors because (1) it doesn't result in too much of overheads and (2) it is quite simple to implement. 

This code is doing exactly the same 

Added comment `Exchanges halo tetrahedrons between MPI neighbors`. See #645